vendor/bat-native-ads/src/bat/ads/internal/time_profiler.h

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
```

Comment:
Another thing: if it's a general mechanism for tracking performance, why it's inside `vendor/bat-native-ads/src/bat/ads/internal/` ?

Reply:
Temporary PR at this time as needed to profile iOS. This PR will not be merged until discussed at a future date, based upon bringing trace event support including UI to iOS